changed-files/action.yml

```
@@ -0,0 +1,36 @@
+name: Changed files
+description: Get list of changed files between two branches
+
+inputs:
+  base-branch:
+    description: Base branch (destination of a pull request)
+    required: true
+    default: ${{ github.base_ref }}
+  head-branch:
+    description: Head branch (source of a pull request)
+    required: false
+    default: ${{ github.head_ref }}
+
+outputs:
+  changed-files:
+    description: List of changed files
+    value: ${{ steps.changed-files.outputs.all_changed_files }}
+
+runs:
+  using: composite
+  steps:
+    - name: Get changed files
+      id: changed-files
+      shell: bash
+      run: |
+        BASE_BRANCH=${{ inputs.base-branch }}
+        HEAD_BRANCH=${{ inputs.head-branch }}
+
+        git fetch origin $BASE_BRANCH --depth 1
+        git fetch origin $HEAD_BRANCH --depth 1
+        CHANGED_FILES=$(git diff --name-only origin/$BASE_BRANCH origin/$HEAD_BRANCH)
+        {
+          echo 'all_changed_files<<EOF'
+          echo "${CHANGED_FILES}"
+          echo EOF
+        } >> $GITHUB_OUTPUT
```

Wouldn't this be a little simpler? I kinda dislike storing multiline strings and then like my docstrings ;)

```suggestion
        cat <<END_OUTPUT >> $GITHUB_OUTPUT
        all_changed_files<<EOF
        $( git diff --name-only origin/$BASE_BRANCH origin/$HEAD_BRANCH )
        EOF
        END_OUTPUT
```

---

TIL of the multi-line output myself :)